### PR TITLE
txscript: handle variable length P2PKH signatures in ComputePkScript

### PR DESCRIPTION
**Pull request by wpaulino**
Since P2PKH signatures have variable lengths, we would attempt to parse P2PKH scripts as P2SH if they didn't fit the previous length constraints.

### CODE REVIEW

File: txscript/pkscript.go [hunk 1]
```diff
@@ -13,13 +13,21 @@ import (
...
-	//   0x47 or 0x48 (71 or 72 byte data push) | <71 or 72 byte sig> |
-	//   0x21 (33 byte data push) | <33 byte compressed pubkey>
-	pubKeyHashSigScriptLen = 106
+	// minPubKeyHashSigScriptLen is the minimum length of a signature script
+	// that spends a P2PKH output. The length is composed of the following:
+	//   Signature length (1 byte)
+	//   Signature without variable R and S values (73 - 33 - 33 bytes)
+	//   Public key length (1 byte)
+	//   Public key (33 byte)
+	minPubKeyHashSigScriptLen = 1 + (73 - 33 - 33) + 1 + 33
```
Comment (Roasbeef) [inline txscript/pkscript.go]: `(73 - 33 - 33)`? So just `7`? That isn't a valid signature length.

Comment (Roasbeef) [reply txscript/pkscript.go]: In place, I think you just want `71` here.

Comment (wpaulino) [reply txscript/pkscript.go]: I've seen cases where the R and/or S values of the signature end up being less than the usual 32 bytes, so the lower end of the range is meant to work around this.

Comment (cfromknecht) [reply txscript/pkscript.go]: minimum signature length is 8: https://github.com/btcsuite/btcd/blob/master/btcec/signature.go#L91

Comment (wpaulino) [reply txscript/pkscript.go]: Ah, that's exactly what I was looking for!

Comment (wpaulino) [reply txscript/pkscript.go]: Fixed.

**Pull request merged**